### ISSUE TITLE
`label-has-for` doesn't pass for dynamic string values

### ISSUE THREAD

**Issue by davidleger95** [evcohen/eslint-plugin-jsx-a11y#330]
This causes a linter error:
```js
<label htmlFor={`InputField-${id}`}>{labelName}</label>
<input id={`InputField-${id}`} name={id} />
```

`id` is a string passed in via props to the parent component.

**Comment by ljharb** [comment]
It should, because the label doesn't wrap the input.

Is that the error message you're getting?

**Comment by davidleger95** [comment]
Why does the label have to wrap the input? Shouldn't `htmlFor === id` pass for this rule?

There error is `Form label must have associated control`

**Comment by ljharb** [comment]
For full a11y coverage, it must *both* be ID-linked and label-wrapped.

See the [docs](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/label-has-for.md): if you want to configure it to require only one method (nesting vs id), or either, you can - but the default is "both".

**Comment by davidleger95** [comment]
Ahhh, that's what I need. Thanks!
